### PR TITLE
Use source model SATNUM regions when generating a FlowNet

### PR DESCRIPTION
**Pull request by olelod**
PR #160 closed using EQLNUM regions from source model, this PR deals with using source model SATNUM regions.

For models where many flow tubes end up outside of the source model simulation grid, looping through all cells to find the closest region can be time consuming. Changed here to find the closest flow tube allready assigned a region, which is much faster (and close to the same).

---

### Contributor checklist

- [x] :tada: This PR closes the last part of #139 
- [x] :scroll: I have broken down my PR into the following tasks:
   - [x] Extract SATNUM regions from input simulation and assign regions to flow tubes
   - [x] Testing
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have considered adding a new entry in `CHANGELOG.md`.
- [ ] :books: I have considered updating the documentation.

### CODE REVIEW

File: .github/workflows/flownet.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
           TESTDATA_REPO_OWNER: equinor
           # If you want the CI to (temporarily) run against another branch than master,
           # change the value her from "master" to the relevant branch name.
-          TESTDATA_REPO_BRANCH: master
+          TESTDATA_REPO_BRANCH: satnum-regions
```
Comment (wouterjdb) [inline .github/workflows/flownet.yml:51]: Replace before merging.

File: src/flownet/ahm/_run_ahm.py [hunk 2]
```diff
@@ -58,24 +59,32 @@ def _from_regions_to_flow_tubes(
...
-                        np.square(tube_midpoint[0] - cell_midpoint[0])
-                        + np.square(tube_midpoint[1] - cell_midpoint[1])
-                        + np.square(tube_midpoint[2] - cell_midpoint[2])
-                    )
+            df_regions.append(None)
+            tube_outside.append(i)
+
+    if tube_outside != []:
+        tube_midpoints = network.get_connection_midpoints()
+        candidate_tubes = set(network.grid.model.unique()) - set(tube_outside)
```
Comment (wouterjdb) [inline src/flownet/ahm/_run_ahm.py:67]: Could you update the function description to describe what this part of the function is doing. It's not really clear to me in one go.

**Review by wouterjdb — CHANGES_REQUESTED** [review]
Looks good @olelod, I just didn't really get a good understanding of a part of the code. If you could clearify then we can merge 👍

File: src/flownet/ahm/_run_ahm.py (hunk 2 above)
Comment (olelod) [reply src/flownet/ahm/_run_ahm.py:68]: Now with updated fucntion description (which should have been done when the code was changed in the first place)

File: src/flownet/ahm/_run_ahm.py [hunk 3]
```diff
@@ -58,24 +60,32 @@ def _from_regions_to_flow_tubes(
...
-                dist_to_cell.append(
-                    np.sqrt(
-                        np.square(tube_midpoint[0] - cell_midpoint[0])
-                        + np.square(tube_midpoint[1] - cell_midpoint[1])
-                        + np.square(tube_midpoint[2] - cell_midpoint[2])
-                    )
+            df_regions.append(None)
+            tube_outside.append(i)
+
+    if tube_outside != []:
```
Comment (anders-kiaer) [inline src/flownet/ahm/_run_ahm.py:66]:
```suggestion
    if tube_outside:
```

File: src/flownet/ahm/_run_ahm.py [hunk 4]
```diff
@@ -58,24 +60,32 @@ def _from_regions_to_flow_tubes(
...
+        tube_midpoints = network.get_connection_midpoints()
+        candidate_tubes = set(network.grid.model.unique()) - set(tube_outside)
+        while len(tube_outside) > 0:
```
Comment (anders-kiaer) [inline src/flownet/ahm/_run_ahm.py:69]:
```suggestion
        while tube_outside:
```

**Review by anders-kiaer — CHANGES_REQUESTED** [review]
Some minor suggestions - otherwise LGTM 🚀

**Review by wouterjdb — APPROVED** [review]
LGTM 🎉

**Review by anders-kiaer — APPROVED** [review]

**Pull request merged**